core/src/main.ts

```diff
@@ -0,0 +1,150 @@
+import { IReporters } from './reporter/types'
+import { buildReporter as buildVrfReporter } from './reporter/vrf'
+import { buildReporter as buildL2VrfFulfillReporter } from './reporter/vrf-L2-fulfill'
+import { buildReporter as buildL2VrfRequestReporter } from './reporter/vrf-L2-request'
+import { CHAIN, REDIS_HOST, REDIS_PORT } from './settings'
+
+const LISTENERS: IListeners = {
+  DATA_FEED: buildDataFeedListener,
+  VRF: buildVrfListener,
+  REQUEST_RESPONSE: buildRequestResponseListener,
+  DATA_FEED_L2: buildL2DataFeedListener,
+  VRF_L2_REQUEST: buildVrfL2RequestListener,
+  VRF_L2_FULFILL: buildVrfL2FulfillListener,
+  REQUEST_RESPONSE_L2_REQUEST: buildRequestResponseL2RequestListener,
+  REQUEST_RESPONSE_L2_FULFILL: buildRequestResponseL2FulfillListener
+}
+
+const WORKERS: IWorkers = {
+  DATA_FEED: dataFeedWorker,
+  VRF: vrfWorker,
+  REQUEST_RESPONSE: requestResponseWorker,
+  DATA_FEED_L2: l2DataFeedWorker,
+  VRF_L2_REQUEST: l2VrfRequestWorker,
+  VRF_L2_FULFILL: l2VrfFulfillWorker,
+  REQUEST_RESPONSE_L2_REQUEST: l2RequestResponseRequestWorker,
+  REQUEST_RESPONSE_L2_FULFILL: l2RequestResponseFulfillWorker
+}
+
+const REPORTERS: IReporters = {
+  DATA_FEED: buildDataFeedReporter,
+  VRF: buildVrfReporter,
+  REQUEST_RESPONSE: buildRequestResponseReporter,
+  DATA_FEED_L2: buildL2DataFeedReporter,
+  VRF_L2_REQUEST: buildL2VrfRequestReporter,
+  VRF_L2_FULFILL: buildL2VrfFulfillReporter,
+  REQUEST_RESPONSE_L2_REQUEST: buildL2RequestResponseRequestReporter,
+  REQUEST_RESPONSE_L2_FULFILL: buildL2RequestResponseFulfillReporter
+}
+
+const LOGGER = buildLogger()
+const FILE_NAME = import.meta.url
+
+async function startListenerService(service: string, redisClient: RedisClientType) {
+  const listenersRawConfig = await getListeners({ service, chain: CHAIN })
+  const listenersConfig = postprocessListeners({
+    listenersRawConfig,
+    service,
+    chain: CHAIN,
+    logger: LOGGER
+  })
+
+  if (!LISTENERS[service] || !listenersConfig[service]) {
+    LOGGER.error({ name: 'main', file: FILE_NAME, service }, 'service')
+    throw new OraklError(OraklErrorCode.UndefinedListenerRequested)
+  }
+
+  LISTENERS[service](listenersConfig[service], redisClient, LOGGER)
+  LOGGER.info('Listener launched')
+}
+
+function startWorkerService(service: string, redisClient: RedisClientType) {
+  WORKERS[service](redisClient, LOGGER)
+  LOGGER.info('Worker launched')
+}
+
+function startReporterService(service: string, redisClient: RedisClientType) {
+  REPORTERS[service](redisClient, LOGGER)
+  LOGGER.info('Reporter launched')
+}
+
+async function main() {
+  hookConsoleError(LOGGER)
+
+  const redisClient: RedisClientType = createClient({ url: `redis://${REDIS_HOST}:${REDIS_PORT}` })
+  await redisClient.connect()
+
+  const service = loadArgs()
+
+  //   start listener, worker, reporter services
+  await startListenerService(service, redisClient)
+  startWorkerService(service, redisClient)
+  startReporterService(service, redisClient)
```

Comment:
is it okay not to await for worker and reporter?

Reply:
yes it should be fine because listener, worker, and reporter are servers and dont need to be awaited. The reason listener is being awaited is because we are making a db call to fetch listener config before we start the listener server. Worker doesn't have a config in the db and reporter config, unlike listener config, is fetched later on down the line